cip/CIP2015-06-24-call-procedures.adoc

```diff
@@ -0,0 +1,250 @@
+= CIP2015-06-24 Calling Stored Procedures
+:numbered:
+:toc:
+:toclevels: 3
+:toc-placement: macro
+:source-highlighter: codemirror
+
+*Author:* Stefan Plantikow <stefan.plantikow@neotechnology.com>
+*Author:* Nigel Small <nigel.small@neotechnology.com>
+*Author:* Petra Selmer <petra.selmer@neotechnology.com>
+*Author:* Rickard Öberg <rickard.öberg@neotechnology.com>
+
+[abstract]
+.Abstract
+
+This proposal introduces stored procedures as a new concept to Cypher and proposes a new facility
+for calling them.
+
+User-defined functions are a separate concern that might be addressed in a future CIP.
+
+toc::[]
+
+== Motivation & Background
+
+While Cypher is a very expressive graph query language, its declarative nature and scope is currently not sufficient for
+expressing certain classes of graph traversals, like traversals relying on using specialized data structures or on accessing third-party functionality.
+
+This CIP proposes to add stored procedures to Cypher in order to address these issues.
+
+== Proposal
+
+This proposal consists of two parts, introducing stored procedures conceptually, and describing how to call them in a
+system that supports Cypher.
+
+=== Stored procedures
+
+A stored procedure has a name and takes a fixed number of named and typed arguments in a fixed sequence and either just
+yields nothing (i.e. only potentially causes a side effect) or yields a stream of result records.
+Each record yielded contains values for a fixed number of named and typed fields in the same fixed sequence such that each field value in a yielded record matches its field's type.
+
+The name of a stored procedure, its argument names, and if applicable any result field names follow the same rules as other symbolic names in Cypher (like variables, labels, relationship types, and property keys).
+
+==== Procedure signatures
+
+The arguments of a procedure, their types, and their order are called its argument signature.
```

Comment:
Can we make 'argument signature' stand out (as it is now a defined term within the context of the CIP)..i.e. use special formatting or quotes?